### PR TITLE
odbcTransport to have an ODBC property

### PR DESCRIPTION
**Pull request by worksofliam**
Needed to specify a CCSID due to some issues with XMLSERVICE.

Please point me in the direction of where I need to update the documentation and I will do that too.

Thanks.

### CODE REVIEW

**Review by GajenderTyagi — APPROVED** [review]
The changes are correct.

**Comment by kadler** [comment]
What was the problem that you're trying to fix and what CCSID were you trying to use?

**Comment by kadler** [comment]
What platform was this on?

Personally I think hard-coding it to 1208 would be better than making it a user-visible option. Node.js wants to work in UTF-8 or UTF-16, so it makes sense to force those encodings. If someone comes up with a reason to need a different encoding, we could make it configurable then.

**Comment by worksofliam** [comment]
This was when running on IBM i.

**Comment by kadler** [comment]
Ah, that makes sense. Other than Windows, IBM i is one of the last modern platforms that defaults to a non-UTF-8 encoding (prior to 7.4).

**Comment by worksofliam** [comment]
So what needs to happen to get this merged?

File: lib/transports/odbcTransport.js [hunk 1]
```diff
@@ -54,6 +55,10 @@ function odbcCall(config, xmlInput, done) {
       connectionString += `PWD=${password};`;
     }
   }
+  
+  if (ccsid) {
+    connectionString += `CCSID=${ccsid};`;
+  }
```
Comment (kadler) [inline lib/transports/odbcTransport.js:61]:
```suggestion
  connectionString += 'CCSID=1208;';
```

File: lib/transports/odbcTransport.js [hunk 2]
```diff
@@ -36,6 +36,7 @@ function odbcCall(config, xmlInput, done) {
     xslib = 'QXMLSERV',
     verbose = false,
     dsn = null,
+    ccsid = null
```
Comment (kadler) [inline lib/transports/odbcTransport.js:39]:
```suggestion
```

**Review by kadler — APPROVED** [review]

**Comment by worksofliam** [comment]
@kadler Plans to merge and release this week?

**Review by abmusse — COMMENTED** [review]
@worksofliam 

Can you clean up the PR Title to follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)?

Looks like our PR linter is failing:
https://github.com/IBM/nodejs-itoolkit/pull/327/checks?check_run_id=2640405302


Something like: `feat: Set CCSID to 1208 for odbc transport` would work

As far as the CI test builds, our build system is currently down so we would have to run the tests manually to confirm tests are still passing.

**Comment by kadler** [comment]
@worksofliam it'd also be good to squash this to one commit

**Comment by kadler** [comment]
For testing, we'd only need to run tests with the ODBC transport.

**Pull request closed without merging**